src/drivers/camera_trigger/camera_trigger.cpp

```diff
@@ -257,12 +293,18 @@ CameraTrigger::start()
 	}
 
 	// enable immediate if configured that way
-	if (_mode > 1) {
+	if (_mode == 2) {
 		control(true);
 	}
 
-	// start to monitor at high rate for trigger enable command
-	work_queue(LPWORK, &_work, (worker_t)&CameraTrigger::cycle_trampoline, this, USEC2TICK(1));
+	// if in distance mode, start monitoring the position
+	if (_mode == 3) {
+		work_queue(LPWORK, &_work, (worker_t)&CameraTrigger::distance_trampoline, this, USEC2TICK(1));
```

Comment:
The issue having different trampolines and triggering them off an action is that you can have active callbacks to both functions at the same time. Do you need a separate function for the triggering? Its generally not a good design pattern.